src/lib/Component.js

```diff
@@ -22,14 +22,25 @@ export default class extends HTMLElement {
 
   #syncAttributesToState() {
     this.state = Array.from(this.attributes).reduce(
-      (state, attr) => ({
-        ...state,
-        [toCamelCase(attr.name)]: attr.value,
-      }),
+      (state, attr) => {
+        const camelCaseName = toCamelCase(attr.name)
+        return {
+          ...state,
+          [camelCaseName]: this.#sanitizeAttribute(camelCaseName, attr.value),
+        }
+      },
       this.state
     )
   }
 
+  #sanitizeAttribute(attrName, attrValue) {
```

Comment:
Can we move this out of class methods?
I guess we could pass `(attrName, attrType, attrValue)` or in a different order.

Reply:
I wasn't sure about this. Updated 👍